lib/ruby_lsp/global_state.rb

```diff
@@ -103,16 +104,22 @@ def encoding_name
 
     private
 
-    sig { params(dependencies: T::Array[String]).returns(String) }
-    def detect_formatter(dependencies)
+    sig { params(direct_dependencies: T::Array[String], all_dependencies: T::Array[String]).returns(String) }
+    def detect_formatter(direct_dependencies, all_dependencies)
       # NOTE: Intentionally no $ at end, since we want to match rubocop-shopify, etc.
-      if dependencies.any?(/^rubocop/)
-        "rubocop"
-      elsif dependencies.any?(/^syntax_tree$/)
-        "syntax_tree"
-      else
-        "none"
+      return "rubocop" if direct_dependencies.any?(/^rubocop/)
+
+      rubocop_is_transitive_dependency = all_dependencies.include?("rubocop")
+      syntax_tree_is_direct_dependency = direct_dependencies.include?("syntax_tree")
+
+      # syntax_tree supports usage together with RuboCop
+      if dot_rubocop_yml_present && rubocop_is_transitive_dependency && !syntax_tree_is_direct_dependency
```

Comment:
I may be missing something here, I believe this already happens that way. Could you show me an example where this currently doesn't work? I think the code would be easier to read if the condition for the direct dependency on `syntax_tree` gets pulled up below the one for the rubocop check.

------

Speaking of tests, I missed that one failure in `server_test.rb`. It's now detecting rubocop since the conditions do apply here now, with also looking at the lockfile specs. I believe most tests against `global_state.formatter` from there are unnecessary and already handled in the global_state test file. The only one I think makes sense is `test_shows_error_if_formatter_set_to_rubocop_but_rubocop_not_available`, the other 4 are redundant. Would you agree?